### PR TITLE
Conceptdsl

### PR DESCRIPTION
**Pull request by renehiemstra**
First implementation of the DSL
* Tests in `test_concepts` have been translated.
* I don't think actual use cases of parameterized concepts work correctly yet. Maybe you can try it out and write a few tests.

### CODE REVIEW

File: test_terraform.t [hunk 1]
```diff
@@ -452,3 +455,209 @@ testenv "terraforming class methods" do
...
+		test [Generic(Special) == true]
+		test [Special(Generic) == false]
+	end
+
+    testset "Multipe inheritance" do
+		local SVec = concepts.parametrizedconcept("SVec")
+
+        concept SVec()
+            Self.traits.length = concepts.traittag
+			Self.methods.length = &Self -> concepts.Integral
```
Comment (tpkessler) [inline test_terraform.t:604]: Can you fix the indentation in this file please?

File: test_terraform.t [hunk 2]
```diff
@@ -452,3 +455,209 @@ testenv "terraforming class methods" do
...
+        end
+
+        concept SVec(T) where {T : concepts.Number}
+			Self:inherit(SVec())
```
Comment (tpkessler) [inline test_terraform.t:608]: Can you rebase against master please? It adds implicit inheritance, so this is not needed anymore.

File: test_terraform.t [hunk 3]
```diff
@@ -452,3 +455,209 @@ testenv "terraforming class methods" do
...
+			Self.methods.axpy = {&Self, T, &Self} -> {}
+        end
+
+        concept SVec(T, N) where {T : concepts.Float, N : 3}
+			assert(N == 3)
+			Self:inherit(SVec(T))
```
Comment (tpkessler) [inline test_terraform.t:614]: See above.

Comment (renehiemstra) [reply test_terraform.t:614]: It looks fine in my editor. Maybe you do it.

Comment (tpkessler) [reply test_terraform.t:614]: I guess it's inconsistent use of tabs and spaces. I'm using tab with 4 spaces.

Comment (tpkessler) [reply test_terraform.t:614]: Never mind. It looks fine locally. So it's github that is confused.

**Pull request merged**